include/ua_securitycontext.h

```diff
@@ -142,6 +143,11 @@ struct UA_Channel_SecurityContext
     UA_StatusCode (*const parseRemoteCertificate)(UA_Channel_SecurityContext *const securityContext,
                                                   const UA_ByteString *const remoteCertificate);
 
+    /**
+     * Gets the signature size. This will return 0 as long as no remote certificate was set.
+     */
+    size_t (*const getSignatureSize)(const UA_Channel_SecurityContext *const securityContext);

```

Comment:
Das dachte ich auch erst. Die Signatur beim symmetrischen verschlüsseln ist in der Tat ein HMAC mit fester Länge. Beim asymmetrischen signieren wird allerdings rsa verwendet. D.h. beim asymmetrischen verschlüsseln ist die der Signatur kleiner gleich die Länge des Schlüssels (also des Modulus (Rsa signatur ist ja m^d mod N, wobei d secret ist und N der öffentliche modulus). Der Schlüssel hat aber variable länge und die kann sich pro Channel unterscheiden und die Signaturlänge hängt somit vom verwendeten Schlüssel ab.
Zumindest habe ich das so verstanden.